keps/sig-node/0009-node-heartbeat.md

```diff
@@ -233,6 +234,17 @@ relying on frequent Node object updates. However, in non-managed solutions, cust
 will still be able to restore previous behavior by setting appropriate flag values.
 Thus, changing defaults to what we recommend is the path to go with.
 
+### Testing Plan
+
+There is a set of dedicated end-to-end tests added for that feature excercising:
+- whether Lease object is being created and update by Kubelet
+- whether Kubelet is reducing frequency of node status updates appropriately
+- whether Lease object is deleted on node deletion
+
+Additionally, if the feature gate is switched on, all existing test suites are
+implicitly testing behavior of this feature, as this is then the signal for
+healthiness of nodes.
```

Comment:
Good question.

There's kubemark:
https://testgrid.k8s.io/sig-scalability-kubemark#kubemark-5000

I don't see instructions for how to interpret the results in the kubemark guide though:
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-scalability/kubemark-guide.md

Reply:
Since I was curious:
Example from latest run: https://storage.googleapis.com/kubernetes-jenkins/logs/ci-kubernetes-kubemark-gce-scale/1511/build-log.txt
Search for "Top latency metric".
```
W0221 01:26:34.024] I0221 01:26:34.024444   22221 api_responsiveness.go:135] APIResponsiveness: Top latency metric: {Resource:nodes Subresource: Verb:LIST Scope:cluster Latency:{Perc50:240.369ms Perc90:295.291ms Perc99:870.811ms} Count:323}; threshold: 10s
W0221 01:26:34.025] I0221 01:26:34.024572   22221 api_responsiveness.go:135] APIResponsiveness: Top latency metric: {Resource:pods Subresource: Verb:LIST Scope:namespace Latency:{Perc50:44.882ms Perc90:51.798ms Perc99:318.453ms} Count:10128}; threshold: 5s
W0221 01:26:34.025] I0221 01:26:34.024584   22221 api_responsiveness.go:135] APIResponsiveness: Top latency metric: {Resource:pods Subresource: Verb:LIST Scope:cluster Latency:{Perc50:156Âµs Perc90:179.876ms Perc99:179.876ms} Count:3}; threshold: 10s
W0221 01:26:34.025] I0221 01:26:34.024592   22221 api_responsiveness.go:135] APIResponsiveness: Top latency metric: {Resource:leases Subresource: Verb:GET Scope:namespace Latency:{Perc50:1.862ms Perc90:8.634ms Perc99:146.831ms} Count:3273182}; threshold: 1s
W0221 01:26:34.026] I0221 01:26:34.024599   22221 api_responsiveness.go:135] APIResponsiveness: Top latency metric: {Resource:pods Subresource: Verb:POST Scope:namespace Latency:{Perc50:1.492ms Perc90:3.788ms Perc99:124.283ms} Count:155016}; threshold: 1s
...
W0221 11:06:04.643] I0221 11:06:04.643028   22221 api_responsiveness.go:135] APIResponsiveness: Top latency metric: {Resource:nodes Subresource: Verb:LIST Scope:cluster Latency:{Perc50:243.395ms Perc90:345.626ms Perc99:1.178885s} Count:1701}; threshold: 10s
W0221 11:06:04.643] I0221 11:06:04.643171   22221 api_responsiveness.go:135] APIResponsiveness: Top latency metric: {Resource:services Subresource: Verb:LIST Scope:cluster Latency:{Perc50:101.436ms Perc90:129.024ms Perc99:405.32ms} Count:399}; threshold: 10s
W0221 11:06:04.644] I0221 11:06:04.643182   22221 api_responsiveness.go:135] APIResponsiveness: Top latency metric: {Resource:pods Subresource: Verb:LIST Scope:namespace Latency:{Perc50:27.837ms Perc90:46.604ms Perc99:229.905ms} Count:46612}; threshold: 5s
W0221 11:06:04.644] I0221 11:06:04.643190   22221 api_responsiveness.go:135] APIResponsiveness: Top latency metric: {Resource:pods Subresource: Verb:DELETE Scope:namespace Latency:{Perc50:4.518ms Perc90:18.402ms Perc99:136.869ms} Count:334634}; threshold: 1s
W0221 11:06:04.644] I0221 11:06:04.643230   22221 api_responsiveness.go:135] APIResponsiveness: Top latency metric: {Resource:services Subresource: Verb:DELETE Scope:namespace Latency:{Perc50:26.928ms Perc90:38.175ms Perc99:135.006ms} Count:8250}; threshold: 1s
```

Reply:
This (for observing no regression on SLI metrics +

and for the gain max size of etcd db size, e.g.:
https://storage.googleapis.com/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-scale-performance/315/artifacts/EtcdMetrics_density_2019-02-22T10:10:08Z.txt
(at the bottom).

I also extended the text.